internal/factsengine/gatherers/corosynccmapctl.go

```diff
@@ -0,0 +1,45 @@
+package gatherers
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/utils"
+)
+
+const (
+	CorosyncCmapCtlFactKey = "corosync-cmapctl"
+)
+
+type CorosyncCmapctlGatherer struct {
+	executor CommandExecutor
+}
+
+func NewCorosyncCmapctlGatherer() *CorosyncCmapctlGatherer {
+	return &CorosyncCmapctlGatherer{
+		executor: Executor{},
+	}
+}
+
+func (s *CorosyncCmapctlGatherer) Gather(factsRequests []FactRequest) ([]Fact, error) {
+	facts := []Fact{}
+	log.Infof("Starting corosync-cmapctl facts gathering process")
+
+	corosyncCmapctl, err := s.executor.Exec( // nolint:gosec
+		"corosync-cmapctl", "-b")
+	if err != nil {
+		return facts, err
+	}
+
+	corosyncCmapctlMap := utils.FindMatches(`(?m)^(\S*)\s\(\S*\)\s=\s(.*)$`, corosyncCmapctl)
+
+	for _, factReq := range factsRequests {
+		if value, ok := corosyncCmapctlMap[factReq.Argument]; ok {
```

Comment:
What if the value is not found?
We should handle this scenario. Maybe by now, we can put a log, but at some point we need to send back a fact with this data (maybe a new define hack).